src/string.rs

```diff
@@ -315,6 +422,261 @@ impl<const N: usize> String<N> {
     /// ```
     #[inline]
     #[allow(clippy::result_unit_err)]
+    pub fn push(&mut self, c: char) -> Result<(), ()> {
+        self.as_mut_view().push(c)
+    }
+
+    /// Shortens this `String` to the specified length.
+    ///
+    /// If `new_len` is greater than the string's current length, this has no
+    /// effect.
+    ///
+    /// Note that this method has no effect on the allocated capacity
+    /// of the string
+    ///
+    /// # Panics
+    ///
+    /// Panics if `new_len` does not lie on a [`char`] boundary.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use heapless::String;
+    ///
+    /// let mut s: String<8> = String::try_from("hello")?;
+    ///
+    /// s.truncate(2);
+    ///
+    /// assert_eq!("he", s);
```

Comment:
```suggestion
    /// assert_eq!(s, "he");
```

Reply:
Done in 6f4bc0698072e178d72cc29e82398d8a69e530b1